include/libobmcsession/manager.hpp

```diff
@@ -0,0 +1,192 @@
+     *
+     * @param userName - the owner user name
+     * @param remoteAddress - the remote ip-address of session creating
+     * initiators'.
+     * @param cleanupFn - the callback to cleanup session on destroy.
+     * @return SessionIdentifier - uniquie session ID
+     */
+    SessionIdentifier create(const std::string& userName,
+                             const uint32_t remoteAddress,
+                             SessionCleanupFn&& cleanupFn);
+
+    /**
+     * @brief Remove a dbus session object from storage and unpublish it from
+     * dbus.
+     *
+     * @param sessionId unique session ID to remove from storage
+     *
+     * @return true - success
+     * @return false - fail
+     */
+    bool remove(SessionIdentifier sessionId);
+
+    /**
+     * @brief Remove all sessions associated with the specified user.
+     *
+     * @param userName - username to close appropriate sessions
+     * @return std::size_t - count of closed sessions
+     *
+     * @throw std::runtime_error not implemented
+     */
+    std::size_t removeAll(const std::string& userName);
+
+    /**
+     * @brief Remove all sessions which have been opened from the specified IPv4
+     * address.
+     *
+     * @param remoteAddress - the IPv4 address to close all appropriate
+     * sessions.
+     *
+     * @throw std::runtime_error not implemented
+     *
+     * @return std::size_t
+     */
+    std::size_t removeAll(uint32_t remoteAddress);
+
+    /**
+     * @brief Remove all sessions of specified type.
+     *
+     * @param type - the type of session to close.
+     *
+     * @throw std::runtime_error not implemented
+     *
+     * @return std::size_t - count of closed sessions
+     */
+    static std::size_t removeAll(SessionType type);
+
+    /**
+     * @brief Unconditional removes all opened sessions.
+     *
+     * @throw std::runtime_error not implemented
+     *
+     * @return std::size_t - count of closed sessions
+     */
+    static std::size_t removeAll();
+
+  protected:
+    friend class SessionItem;
+
+    /**
+     * @brief Generate new session depened on current session manager slug and
+     * timestamp.
+     *
+     * @return SessionIdentifier - a new session identifier
+     */
+    SessionIdentifier generateSessionId() const;
+
+    /**
+     * @brief Get the Session Object Path object
+     *
+     * @return const std::string - object path of session for specified
+     * identifier
+     */
+    const std::string getSessionObjectPath(SessionIdentifier) const;
+
+    /**
+     * @brief Get the Session Manager Object Path object
+     *
+     * @return const std::string - object path of session manager for specified
+     * identifier
+     */
+    const std::string getSessionManagerObjectPath() const;
+
+    /**
+     * @brief Cast session identifier to the hex view.
+     *
+     * @return const std::string - a hex view of session identifier.
+     */
+    static const std::string hexSessionId(SessionIdentifier);
```

Comment:
looks like a candidate for an assignment operator override. I don't insist though.

Reply:
Currently, I've kept it as is.